src/planwise/client/current_project/components/sharing.cljs

```diff
@@ -0,0 +1,88 @@
+(ns planwise.client.current-project.components.sharing
+  (:require [re-frame.core :refer [subscribe dispatch]]
+            [re-com.core :as rc]
+            [reagent.core :as r]
+            [clojure.string :as str]
+            [leaflet.core :refer [map-widget]]
+            [planwise.client.asdf :as asdf]
+            [planwise.client.components.common :as common]
+            [planwise.client.utils :as utils]
+            [planwise.client.styles :as styles]))
+
+(defn- search-box
+  []
+  (let [search-string (subscribe [:current-project/shares-search-string])]
+    (fn []
+      [:div.small-search-box
+       [:input
+        {:type "search"
+         :placeholder "Search"
+         :value @search-string
+         :on-change (utils/dispatch-value-fn :current-project/search-shares)}]])))
+
+(defn share-dialog []
+  (let [view-state (subscribe [:current-project/view-state])
+        share-link (subscribe [:current-project/share-link])
+        project-shares (subscribe [:current-project/shares])
+        filtered-shares (subscribe [:current-project/filtered-shares])
+        emails-text (subscribe [:current-project/sharing-emails-text])
+        send-emails-action (subscribe [:current-project/sharing-send-emails])
+        share-token-state (subscribe [:current-project/sharing-token-state])]
+    (fn []
+      (let [close-fn #(dispatch [:current-project/close-share-dialog])
+            key-handler-fn #(when (= 27 (.-which %)) (close-fn))
+            select-text-fn #(.select (.-target %))
+            reset-share-token-fn (utils/with-confirm
+                                   #(dispatch [:current-project/reset-share-token])
+                                   (str "Are you sure you want to reset the sharing link?\n\n"
+                                         "Anyone who has not yet opened the link will not "
+                                         "be able to access, and will need to be shared the new "
+                                         "link."))]
```

Comment:
The wording here is a bit weird. How about "If you reset the link, anyone who has not yet accessed the project will not be able to do it until you invite them again."